File: MiniDOMBase/DMD/Sistema/Users/CUsersComparer.vb
Imports System.Net.Mail 'importo il Namespace
Imports System.Net.Sockets
Imports minidom
Imports minidom.Databases
Imports minidom.Anagrafica

Partial Public Class Sistema

    Public Class CUsersComparer
        Implements IComparer

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub

        Public Function Compare(ByVal x As Object, ByVal y As Object) As Integer Implements IComparer.Compare
            Dim a As CUser = x
            Dim b As CUser = y
            Return Strings.Compare(a.Nominativo, b.Nominativo, CompareMethod.Text)
        End Function

    End Class


End Class

File: MiniDOMBase/DMD/Sistema/Users/Comparers/CUsersComparer.cs
using System.Collections;
using System.Collections.Generic;

namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Compare di oggetti <see cref="CUser"/>
        /// </summary>
        public class CUsersComparer 
            : IComparer, IComparer<CUser>
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public CUsersComparer()
            {
                //DMDObject.IncreaseCounter(this);
            }

            /// <summary>
            /// Compare due oggetti
            /// </summary>
            /// <param name="a"></param>
            /// <param name="b"></param>
            /// <returns></returns>
            public int Compare(CUser a, CUser b)
            {
                return DMD.Strings.Compare(a.Nominativo, b.Nominativo, true);
            }

            int IComparer.Compare(object x, object y)
            {
                return this.Compare((CUser)x, (CUser)y);
            }
        }
    }
}